java/jdbc/SpringBootSample/src/main/java/com/oracle/springapp/DataSourceConfig.java

```diff
@@ -0,0 +1,72 @@
+package com.oracle.springapp;
+
+import java.sql.SQLException;
+
+import javax.sql.DataSource;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+
+import oracle.ucp.jdbc.PoolDataSource;
+import oracle.ucp.jdbc.PoolDataSourceFactory;
+
+/**
+ * This will serve as your primary data source
+ * If you do not want to use this as primary datasource remove the @Primary
+ * and if you want to use OracleUniversalConnectionPool you can add @Qualified
```

Comment:
As UCP is used, shouldn't we say "If you don't want to use UCP ...remove @Qualified"?

Reply:
If the customers don't want to use UCP then, they don't even need this DataSourceConfig.java. Spring deduces that it is Oracle JDBC connection based on the URL read from the application.properties file.

Reply:
If so, why add the comment?
" ... and if you want to use OracleUniversalConnectionPool you can add @Qualified"
Is there a risk of conflict?